### DISCUSSION TITLE
How would I approach implementing `you like this post` feature?

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#3311] [Q&A] [answered]
Consider the following schema:

```prisma
model User {
  id Int @id @default(autoincrement())
  name String
  likes Post[] @relation(references: [id])
}

model Post {
  id Int @id @default(autoincrement())
  title String
  likedBy User[] @relation(references: [id])
}
```

Suppose I need to tell the user he already liked a list of posts, in SQL I would do something like this:

```sql
SELECT id, title
        , EXISTS(
                SELECT *
                FROM likes lk
                JOIN users u ON u.id = lk.user_id AND lk.post_id = p.id
                WHERE u.name = 'john'
                ) AS liked
FROM posts p
        ;
```

What would be the equivalent using prisma?

**Comment by ryands17** [comment, marked as answer]
Hey @brielov 👋 
Something like this should work :)

```ts
const res = await prisma.user.findOne({
    where: {
      id: 1,
    },
    include: {
      likes: true,
    },
  })
```

**Reply by brielov** [reply]
That would return likes as a list of posts, I would rather return a list of posts with a boolean field `liked` if it is liked by the current user

**Reply by ryands17** [reply]
Hey @brielov 👋 
I don't think this is supported with Prisma yet, but there are a couple of workarounds that you can perform:

1. Use the same raw query above via `prisma.queryRaw`

2. Use the following query and liked would be a condition on the frontend (`userId === likedBy[0].id`):

```ts
const posts = await prisma.post.findMany({
    select: {
      title: true,
      likedBy: {
        select: { id: true, name: true },
        where: {
          id: 1,
        },
      },
    },
  })
```

**Reply by brielov** [reply]
Fair enough, I think I'll go with the raw query for now and upgrade in the future if this use case is made easier. Thank!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
